starlette/convertors.py

```diff
@@ -1,67 +1,66 @@
 import math
-import typing
+from typing import Any, Dict, Generic, TypeVar
 
+T = TypeVar("T")
 
-class Convertor:
-    regex = ""
 
-    def convert(self, value: str) -> typing.Any:
+class Convertor(Generic[T]):
+    regex = r""
+
+    def convert(self, value: str) -> T:
         raise NotImplementedError()  # pragma: no cover
 
-    def to_string(self, value: typing.Any) -> str:
+    def to_string(self, value: Any) -> str:
         raise NotImplementedError()  # pragma: no cover
 
 
-class StringConvertor(Convertor):
-    regex = "[^/]+"
+class StringConvertor(Convertor[str]):
+    regex = r"[^/]+"
 
-    def convert(self, value: str) -> typing.Any:
+    def convert(self, value: str) -> str:
         return value
 
-    def to_string(self, value: typing.Any) -> str:
+    def to_string(self, value: Any) -> str:
         value = str(value)
         assert "/" not in value, "May not contain path separators"
         assert value, "Must not be empty"
         return value
 
 
-class PathConvertor(Convertor):
-    regex = ".*"
-
-    def convert(self, value: str) -> typing.Any:
-        return str(value)
+class PathConvertor(StringConvertor):
```

Comment:
I'd probably prefer that we *don't* subclass `StringConvertor` here.

Subclassing `Convertor` instead is a simpler, flatter, style that simply declares "we implement the (stubbed-out) Convertor interface".

Subclassing `StringConvertor` instead adds an indirection, when we could just implement a trivial `convert` method explicitly.